### DISCUSSION TITLE
"GitHub CoPilot has not been started" problem

### DISCUSSION THREAD

**Discussion by amirdar4** [community/community#61137] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi all,

I got a problem with CoPilot in visual studio 2022
when I hover over the icon I get this prompt "GitHub CoPilot has not been started"

VS 2022 version is Version 17.6.5
I'm currently using the free trial
I'm logged in with my company's email and I have connected my account in GitHub to my VS

after logout and login again this is what i get in the output view:

13:53:39:853	GitHub Copilot Agent v1.94.0
13:53:46:506	[DEBUG] [agent] [2023-07-18T10:53:46.495Z] Agent service starting
13:53:46:762	[DEBUG] [agent] [2023-07-18T10:53:46.663Z] Telemetry initialized
13:53:47:891	Auth Status: OK
13:53:48:604	Successfully authorized GitHub Copilot.
13:53:48:606	[INFO] Attempted sign in activeAccount https://github.com/122036912
14:00:26:815	Auth Status: NotSignedIn
14:01:19:296	Auth Status: OK
14:01:19:326	Successfully authorized GitHub Copilot.
14:01:19:327	[INFO] Attempted sign in activeAccount https://github.com/122036912

I'm not sure what is this "122036912". when I press on the line "https://github.com/122036912" I get a 404 page from GitHub domain

I've uninstalled and reinstalled the extension

nothing seems to work 

Any suggestions?

**Comment by andersonrocha** [comment]
I'm experiencing the same problem today.

**Reply by ayim** [reply]
Hi @andersonrocha sorry to hear this is happening, we'd love to diagnose and fix.

Would you mind filing a bug for us to take a look at?

1) Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2) Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3) Run the command devenv /log (this will launch Visual Studio with verbose logging)
4) Use the report a problem function to create a ticket
5) Copy all the contents of the Github Copilot Output Window and add them to the ticket filed.

Should you have any issues here, feel free to email me at ayim at github dot com to schedule a call

**Comment by ion-mironov** [comment]
I'm having the same problem now. My GitHub account was already registered with VS 2022 Community Edition. And, when I click on the link provided in the Output box:

```
Successfully authorized GitHub Copilot.
[INFO] Attempted sign in activeAccount https://github.com/97184530
```
I'm greeted with a 404 page.


**Update**: Okay, this is weird. It continues to say, "_GitHub Copilot has not been started_" when I hover the mouse pointer over the little icon in the lower left, but it appears to be actually working. I am definitely getting inline suggestions as I type (it's definitely GitHub Copilot's suggestions because it's the same  format as what I've gotten in Visual Studio Code). Also, when I do the keyboard shortcut (Ctr+Alt+\) or manually select **Trigger Inline Suggestion**, the inline suggestion pops up.

**Reply by amirdar4** [reply]
@IonMironov-main thx for the input. i double checked and I also get suggestions in my code

**Reply by BrianJSavage** [reply]
Same for me

**Reply by ayim** [reply]
Folks @BrianJSavage @amirdar4 @IonMironov-main, would you mind filing a bug for us to help us resolve this issue?

1) Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2) Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3) Run the command devenv /log (this will launch Visual Studio with verbose logging)
4) Use the report a problem function to create a ticket
5) Copy all the contents of the Github Copilot Output Window and add them to the ticket filed.

Should you have any issues here, feel free to email me at ayim at github dot com to schedule a call

**Comment by la-cc** [comment]
I had the same issue before. 
I tried various things, and the simplest solution was to perform a hard logout from the account and then sign back in. 

This resolved the problem for me. 

Here is a screenshot of the "Sign Out" process.

![sign_out](https://github.com/community/community/assets/110809333/6d28ed91-1a21-4ff9-a473-31592ecbd7b5)

**Reply by amirdar4** [reply]
thanks for the advice
but it didn't work in VS2022/2019

**Comment by Akash1134** [comment]
Hi Everyone,

This issue has already been highlighted to our engineering team and we are working towards a fix. 

Thanks

**Reply by ayim** [reply]
Update: we've root caused the issue, working on a fix.
Will update again when we have an ETA.

**Comment by micktion** [comment]
@Akash1134 is there an update on this?  It actually seems like the Copilot extension in Visual Studio 2022 is working, at least partially, despite the rollover saying it hasn't been started.  What's the impact of this issue?  An ideas on when it'll be resolved?

**Reply by ayim** [reply]
@micktion product owner for the extension here - it appears that this generally affects users who are authenticating for the first time. If it's working for you... let us know if anything changes.

**Reply by micktion** [reply]
@ayim I had used the pre-release version of Copilot and have recently purchased a personal license.  

It's hard to say if it's working or not, I'm seeing output being generated in the Copilot output console like it is attempting to produce suggestions, however it might just be intellisense producing the actual suggestions appearing in the editor.  

The rollover on the copilot icon, bottom left corner of editor, still says "Github Copilot has not been started"

When I look at the extension it's saying Pricing Category: Trial

![image](https://github.com/community/community/assets/4677284/cd3257e1-6af8-4740-96ea-26f5f102c083)

Is that expected?

**Reply by juihanamshet1** [reply]
@micktion Hi Mick, one easy way to check if its working or not is - create a new class and add a comment inside `// method to add two numbers`, when you hit enter to trigger a completion, copilot should return an Add method. If it still doesn't work, please send us your logs. 

1. Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2. Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3. Run the command devenv /log (this will launch Visual Studio with verbose logging)
4. Use the report a problem function to create a ticket
5. Copy all the contents of the Github Copilot Output Window and add them to the ticket filed.

**Comment by arjen1981** [comment]
I am still experiencing the same issue in Visual Studio 2022. I have reinstalled the extensions and VS2022 is updated to the latest version. Sign out and signing in again does not seem to work either. I just discovered this thread.  I am using an individual account and not a business account. So I can confirm that it isn't working yet.

**Reply by CarlosHp1996** [reply]
I have the same problem here...

**Reply by ayim** [reply]
@arjen1981 @CarlosHp1996 would you mind filing a bug for us to help us resolve this issue?

1) Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2) Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3) Run the command devenv /log (this will launch Visual Studio with verbose logging)
4) Use the report a problem function to create a ticket
5) Copy all the contents of the Github Copilot Output Window and add them to the ticket filed.

Should you have any issues here, feel free to email me at ayim at github dot com to schedule a call

**Reply by CarlosHp1996** [reply]
@ayim I have this message in visual studio and when I access the link, I get a 404 error, do you know how to solve it? my visual studio is up to date and I already logged out and logged in from the github account in visual studio, without success....My subscription is active.

![image](https://github.com/community/community/assets/108143798/b499fed6-8fc1-468d-af28-0c0b8c536731)

![image](https://github.com/community/community/assets/108143798/c9cc038f-390f-4a51-b88c-4a4021ea5235)

**Comment by cheema-corellian** [comment]
Sam problem for me.  I get this message, but GitHub Copilot appears to provide code suggestions.

**Reply by ayim** [reply]
@cheema-corellian would you mind filing a bug for us to help us resolve this issue?

1) Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2) Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3) Run the command devenv /log (this will launch Visual Studio with verbose logging)
4) Use the report a problem function to create a ticket
5) Copy all the contents of the GitHub Copilot Output Window and add them to the ticket filed.

Should you have any issues here, feel free to email me at ayim at github dot com to schedule a call

**Comment by redapplewithleaves** [comment]
My team is having this issue too. We've paid over a month for a license already, haven't gotten a single suggestion for copilot

**Reply by wsrinivas** [reply]
Same here mate

**Reply by ayim** [reply]
@redapplewithleaves @wsrinivas  we'd love to get you to a much happier place.
Would you mind filing a bug for us to help us resolve this issue?

1) Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2) Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3) Run the command devenv /log (this will launch Visual Studio with verbose logging)
4) Use the report a problem function to create a ticket
5) Copy all the contents of the GitHub Copilot Output Window and add them to the ticket filed.

Should you have any issues here, feel free to email me at ayim at github dot com to schedule a call

**Comment by Romo031** [comment]
Same thing here, wondering if an ETA could be given soon on when we can expect a fix.

**Comment by ChillstepCoder** [comment]
Just purchased license, installed extension, when I run VS 2022 it never pops up for authentication, just throws these messages. Unfortunate that there are no settings at all on the extension and seemingly no way to do anything about it. Using individual license.

GitHub Copilot 1.94.0.0 (v1.94.0.0@6586dd2df)
GitHub Copilot Agent v1.94.0
[DEBUG] [agent] [2023-08-04T18:36:36.740Z] Agent service starting
[DEBUG] [agent] [2023-08-04T18:36:38.448Z] Telemetry initialized
[INFO] [auth] [2023-08-04T18:36:38.876Z] Invalid copilot token: missing token: 404 
Auth Status: NotAuthorized
[INFO] [auth] [2023-08-04T18:36:39.670Z] Invalid copilot token: missing token: 404 
Auth Status: NotSignedIn
[INFO] [auth] [2023-08-04T18:36:39.874Z] Invalid copilot token: missing token: 404 
[INFO] [auth] [2023-08-04T18:36:39.990Z] Invalid copilot token: missing token: 404 
Auth Status: NotAuthorized
You don't have access to GitHub Copilot.
[INFO] Attempted sign in activeAccount https://github.com/37784165

**Reply by ayim** [reply]
Hi @ChillstepCoder, the extension uses the GitHub account that you're signed into Visual Studio with.
Either this account doesn't have a Copilot entitlement, or you can add the GitHub account with the Copilot entitlement.
i.e.
![image](https://github.com/community/community/assets/10951869/32020599-127b-456f-b660-96a547220eeb)

**Reply by Delayghty** [reply]
I have the exact same Problem. I purchased license, installed extension yesterday on VS C 2022 17.6.4.
I checked what ayim said and added my GitHub Account as his screenshot is showing.
Makes no difference...

_Nevertheless, I found out how I can at least successfully authenticate myself.
I went to "Extensions" > "Manage Extensions" in the menu and then to "Change settings for extensions" at the bottom.
In the new window I went to "Github" > "Copilot" in the left column and set "Sign in with Device Flow" to "True"._

The authentication works now, but Copilot keeps saying "Github copilot has not been started"
<img width="345" alt="image" src="https://github.com/community/community/assets/106477134/e06506c5-95e7-4cfd-8963-5635064e5549">

**Reply by CarlosHp1996** [reply]
Same problem here...

**Reply by juihanamshet1** [reply]
@ChillstepCoder, @Delayghty, and @CarlosHp1996 

If you're seeing this log `Invalid copilot token: missing token: 404`, try the following: 
1. Remove your Github account from VS, close all VS windows. Open VS, sign into Github in VS and check if Copilot works. 
2. If Step 1 didn't work and you still get a 404 then try this. Go to Tools -> Options -> Copilot and set "Sign in with device flow" to True. Restart VS. This time when you click on the Copilot icon, you will the option "Sign in with device flow". Follow the prompt. 

If you're seeing the "GitHub Copilot has not been started" message, ignore it and check if Copilot works anyways. One easy way to check is to create a new class and type `// method to add two numbers`. Hit enter to trigger a Copilot suggestion and Copilot will suggest an Add method. 

If you want to use the chat functionality, install the [GitHub Copilot chat extension](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot)

If you're still facing issues with Copilot, please file a VS feedback ticket. 

1. Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2. Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3. Run the command `devenv /log` (this will launch Visual Studio with verbose logging)
4. Use the report a problem function to create a ticket
5. Copy all the contents of the GitHub Copilot Output Window and add them to the ticket filed.

**Comment by Spixmacaw** [comment]
This error indicates that you either do not have a GitHub Copilot subscription, or there was an error connecting to the GitHub API to request a token to use GitHub Copilot.

**Comment by juihanamshet1** [comment, marked as answer]
Hi folks, we released a fix for the problem where Copilot icon was showing "GitHub Copilot has not been started". Please update your [extension ](https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs&ssr=false#overview) and let us know if you run into further issues! 

If you're still facing issues with Copilot, please file a VS feedback ticket.

1. Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
2. Report a problem from VS2022: https://aka.ms/vs/report-a-problem
3. Run the command devenv /log (this will launch Visual Studio with verbose logging)
4. Use the report a problem function to create a ticket
5. Copy all the contents of the GitHub Copilot Output Window and add them to the ticket filed.

**Reply by andersonrocha** [reply]
Even with device flow enabled the problem still persists. Here's the Github Copilot prompt from VS:

GitHub Copilot 1.100.0.0 (v1.100.0.0@6ff082509)
GitHub Copilot Agent v1.100.0
[DEBUG] [agent] [2023-08-09T17:09:16.762Z] Agent service starting
[DEBUG] [agent] [2023-08-09T17:09:16.820Z] Telemetry initialized
[INFO] [auth] [2023-08-09T17:09:17.197Z] Invalid copilot token: missing token: 403 
Auth Status: NotAuthorized
[INFO] [auth] [2023-08-09T17:09:17.475Z] Invalid copilot token: missing token: 403 
Auth Status: NotSignedIn
NotSignedIn
NotSignedIn
NotSignedIn
NotSignedIn
Please open https://github.com/login/device and enter the device activation code EA92-C287 that has been copied to the clipboard.
[INFO] [auth] [2023-08-09T17:09:59.907Z] Invalid copilot token: missing token: 403 
Auth Status: NotAuthorized
You don't have access to GitHub Copilot.

**Reply by juihanamshet1** [reply]
Hi @andersonrocha, looks like your issue is different because you don't have access to Copilot. Please file a separate issue for that and we'll look into it.

**Comment by micktion** [comment]
As of today the status is now correctly displaying for me...

![image](https://github.com/community/community/assets/4677284/10ac65cf-a315-42b3-bbee-9ace519a010d)

and no longer saying "GitHub CoPilot has not been started"

On startup Copilot output window displays....
`
GitHub Copilot 1.100.0.0 (v1.100.0.0@6ff082509)
GitHub Copilot Agent v1.100.0
[DEBUG] [agent] [2023-08-10T02:54:50.177Z] Agent service starting
[DEBUG] [agent] [2023-08-10T02:54:50.249Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.
`

**Comment by amirdar4** [comment]
Problem is now fixed. thx

**Comment by beyro** [comment]
I am facing the same issue.
Icon says it has not started even though the logs show I authenticated okay.
Verbose logging shows an error getting local issuer certificate "unable to get local issuer certificate" - maybe because ZScaler is running on my machine?

I created an issue:
https://developercommunity.visualstudio.com/t/GitHub-CoPilot-extension-not-working/10437557?port=1026&fsid=f2dadc4a-25cc-427e-a503-22a40c6b4f3b
